src/web/lib/components/ThemeCustomBackgroundPicker/index.js

```diff
@@ -23,6 +24,37 @@ export class ThemeCustomBackgroundPicker extends React.Component {
     super(props);
   }
 
+  componentDidUpdate(prevProps) {
+    const { themeCustomBackgrounds } = this.props;
+
+    // sync images in local storage with custom backgroundsin preview on undo/redo
+    if (prevProps.themeCustomBackgrounds.length !== themeCustomBackgrounds.length) {
```

Comment:
This condition (and the conditions below that also rely on lengths) only works if an item is added, removed or re-ordered.
If an item is replaced, or if multiple items are added/removed and re-ordered, then the logic here may not work as expected (i.e. images being added/removed while they should not be).

Are these length-conditions correct?